### ISSUE TITLE
Try DynamicMethod with ref locals/return scenarios

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#17993]
We expect that DynamicMethod should already handle ref locals, since it operates at the IL level.
We're not sure whether it allows declaring ref return methods.
Either way, it would be good to exercise those code paths to make sure they work.

FYI @VSadov

**Comment by weltkante** [comment]
> We're not sure whether it allows declaring ref return methods.

Nope, they don't work, just have been running into this problem while prototyping some IL generator code. It works when using DynamicAssembly, but DynamicMethod throws an exception when you try to use a ref type as returned type.
